compiler/crates/graphql-ir/src/ir.rs

```diff
@@ -326,7 +326,7 @@ impl Condition {
 // Associated Types
 
 /// @ Name Arguments?
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
Just out of curiosity. Why do we need to derive `Hash` for these structs?

Reply:
Started at `InlineDirectiveMetadata` needing `Hash`. Going down the chain from `InlineDirectiveMetadata` => `VariableDefinition` => `Directive`